### PR TITLE
MINOR: Improve FK Join docs and optimize null-fk case

### PR DESCRIPTION
**Pull request by vvcephei**
* Fix the formatting and wording of the foreign-key join javadoc
* Optimize handling of `null` extracted foreign keys

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by vvcephei** [comment]
Hey all,

I know it's a little awkward to tack an optimization onto a javadoc update, but while I was clarifying the javadocs around what happens when the `foreignKeyExtractor` returns null, I realized we can short-circuit a lot by just immediately dropping the update.

For one thing, we shouldn't be calling the foreign key serializer with a null value anyway, and now we won't.

Note, this doesn't change any behavior. We would previously have sent the update to a repartition topic with a `null` key and then the processor on the other side of the repartition (in `SubscriptionStoreReceiveProcessorSupplier`) immediately checks if the key is null, logs the identical warning, updates the sensor, and drops the update.

**Review by bbejeck — APPROVED** [review]

**Comment by bbejeck** [comment]
ping either of @guozhangwang @mjsax for a second review

**Review by ableegoldman — APPROVED** [review]
This LGTM

**Comment by bbejeck** [comment]
Merged 7536 into trunk

**Comment by bbejeck** [comment]
cherry-picked to 2.4

File: streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java [hunk 1]
```diff
@@ -2120,19 +2120,20 @@
...
+     * @param foreignKeyExtractor a {@link Function} that extracts the key (KO) from this table's value (V). If the
+     *                            result is null, the update is ignored as invalid.
+     * @param joiner              a {@link ValueJoiner} that computes the join result for a pair of matching records
+     * @param named               a {@link Named} config used to name the processor in the topology
+     * @param materialized        a {@link Materialized} that describes how the {@link StateStore} for the resulting {@code KTable}
+     *                            should be materialized. Cannot be {@code null}
+     * @param <VR>                the value type of the result {@code KTable}
+     * @param <KO>                the key type of the other {@code KTable}
+     * @param <VO>                the value type of the other {@code KTable}
      * @return
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java:2137]: @vvcephei we should fix the return statement -- should not be empty

**Pull request merged**